tests/Unit/ProxyGeneratorTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Zenstruck\Foundry\Tests\Unit;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Zenstruck\Foundry\Configuration;
+use Zenstruck\Foundry\ProxyGenerator;
+use Zenstruck\Foundry\Tests\Fixtures\Entity\Post;
+use Zenstruck\Foundry\Tests\Fixtures\Entity\Tag;
+
+class ProxyGeneratorTest extends TestCase
+{
+    /** @var ObjectManager|MockObject */
+    private $objectManager;
+    /** @var ManagerRegistry|MockObject */
+    private $managerRegistry;
+    /** @var Configuration|MockObject */
+    private $config;
+    /** @var ProxyGenerator */
+    private $generator;
+
+    protected function setUp(): void
+    {
+        $this->objectManager = $this->createMock(ObjectManager::class);
+        $this->managerRegistry = $this->createMock(ManagerRegistry::class);
+        $this->config = new Configuration();
+        $this->config->setManagerRegistry($this->managerRegistry);
+        $this->generator = new ProxyGenerator($this->config);
+    }
+
+    /** @test */
+    public function can_auto_refresh_all_methods(): void
+    {
+        $tag = new Tag();
+        $this->managerRegistry->expects($this->any())->method('getManagerForClass')->with(Tag::class)->willReturn($this->objectManager);
+        $proxyTag = $this->generator->generate($tag);
+
+        $this->objectManager->expects($this->exactly(2))->method('refresh')->with($tag);
+        $proxyTag->setName('proxy');
+        $this->assertEquals('proxy', $proxyTag->getName());
+    }
+
+    /** @test */
+    public function can_only_auto_refresh_getters(): void
+    {
+        $tag = new Tag();
+        $this->managerRegistry->expects($this->any())->method('getManagerForClass')->with(Tag::class)->willReturn($this->objectManager);
+        $proxyTag = $this->generator->generate($tag, ['get*']);
```

Comment:
This is my try at "working around" the auto-refresh problems from #113. You can specify which methods should be proxied/use auto-refresh (allowing for `*` globs, like `get*` or `['get*', 'has*, 'is*']`). The idea is that you will be able to configure this in the model factory.

I'm not sure if this is the ideal solution, but I didn't really understand the `spl_object_id()` solution proposed in #113 😇

Reply:
I can't tell if this solves the issue in #113. Might be better to test on an entity with multiple fields like `Post`.

> but I didn't really understand the spl_object_id()

Long story short, I don't think this will work. Basically, I was trying to come up with a way to only refresh the object if it's been changed in the db.

Reply:
Basically, my idea was that a user can limit which methods will cause a refresh. E.g. you wouldn't necessarily refresh on setters, which automatically resolves the issue when calling multiple setters.

We can maybe also make this more "magic" by running the fresh post-method invocation and only run it for methods that return something (and then, after a refresh, call the method again on the entity from within the interceptor). However, not sure if there are edge-cases there (e.g. fluent setters).

Reply:
I had thought about limiting to non-setters (and of course make configurable) but for models that have rich domains, this becomes a burden I think. I came to the conclusion that it's better to refresh everything and give the ability to disable auto-refreshing for certain circumstances which I believe are rare.

I'm relying on auto-refreshing (on by default) in 100's of tests and only have one instance where I have to disable it.